cypress/integration/rancid_tomatillos_spec.js

```diff
@@ -84,11 +90,61 @@ describe('Movie details page', () => {
       .get('p').contains('99 min')
       .get('p').contains('change her life forever')
       .get('button')
-  })
+  });
 
   it('Should be able to travel back to the home page', () => {
     cy.get('.nav-bar__home').click()
       .location('pathname').should('eq', '/')
-  })
+  });
+});
+
+describe('Error handling', () => {
+  it('Should display an error message when visiting a nonexistent URL', () => {
+    cy.fixture('mockData').then(( data ) => {
+      cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+      body:  data
+      })
+    })
+      .visit('http://localhost:3000/foo')
+      .get('.error__message')
+      .contains('Something went wrong')
+  });
+
+  it('Should display a specific error message when fetch yields a 404 status', () => {
+    cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+      statusCode: 404,
+    })
+      .visit('http://localhost:3000/')
+      .get('.error__message')
+      .contains('Page not found')
+  });
+
+  it('Should display a specific error message when fetch yields a 500 status', () => {
+    cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+      statusCode: 500,
+    })
+      .visit('http://localhost:3000/')
+      .get('.error__message')
+      .contains('Please try again later')
+  });
+
+  it('Should display an error message when there is a network error', () => {
+    cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+      forceNetworkError: true
+    })
+      .visit('http://localhost:3000/')
+      .get('.error__message')
+      .contains('Something went wrong')
+  });
 
-})
\ No newline at end of file
+  it('Should display error message if network down and clicked on movie poster', () => {
+      cy.intercept('https://rancid-tomatillos.herokuapp.com/api/v2/movies/632618', {
+        forceNetworkError: true
+      })
```

Comment:
this is like a status code of 500?

Reply:
No, I think it’s different. I think this is catching the case where the server is completely inaccessible, like if its power goes out. Not 100% on that though. I’d like to learn more about this in general.